### ISSUE TITLE
Error: Error reading input stream at line 167563: Invalid character ( ) in sequence

### ISSUE THREAD

**Issue by antonkulaga** [bbuchfink/diamond#179]
When I build an index on naked_mole_rat proteins http://alfred.liv.ac.uk/downloads/naked_mole_rat/naked_mole_rat_proteins.zip I get errors because of () in sequences names. I think it should be ok to have brackets.

**Comment by bbuchfink** [comment]
The problem is at line 167563 in your file, which contains this line within
a sequence:

Resource temporarily unavailable (4).

**Comment by antonkulaga** [comment]
@bbuchfink just check, it download fine. And at line 167563 I deleted () but it did not help. I suspect it is just a diamond bug

**Comment by bbuchfink** [comment]
I didn't mean the download. This is what your file looks like, containing
an invalid string within the sequence. You need to delete this whole line
not only the ().

>gi|512998478|ref|XP_004860795.1| PREDICTED: exocyst complex component 7
isoform X4 [Heterocephalus glaber]
MIPPQEASARRREIEDKLKQEEETLSFIRDSLEKSDQLTKNMVSILSSFESRLMKLENSIIPVHKQTENL
QRLQENVEKTLSCLDHVISYYHVASDTEKIIREGPTGRLEEYLGSMAKIQKAVEYFQDNSPDSPELNKVK
LLFERGKESLESEFRSLMTRHSKVVSPVLILDLISGDEELEVQEDVALEHLPESVLQDVVRISLWLVEYG
RNQDFMNVYYQIRSSQLDRSIKGLKEHFRKSSSSSGVPYSPAIPNKRKDTPTKKPIKRPGHEHDFRVKHL
SEALNDKHGPLPGRDDMLDVETDAYIHCVSAFVRLAQSEYQLLMDIIPEHHQKKTFDSLIQDALDGLMLE
GENIVSAARKAIIRHDFSTVLTVFPILRHLKQTKPEFDQVLQGTAASTKNKLPGLITSMETIGAKALEDF
ADNIKNDPDKEYNMPKDGTVHELTSNAILFLQQLLDFQETAGAMLASQETSSSATSYSSEFSKRLLSTYI
CKVLGNLQLNLLSKSKVYEDPALSAIFLHNNYNYILKSLEKSELIQLVAVTQKTAERSYREHIEQQIQTY
QRSWLKVTDYISEKNLPVFQPGVKLRDKERQMIKERFKGFNDGLEELCKIQKAWAIPDTEQRDKIRQAQK
NIVKETYGAFLHRYSSVPFTKNPEKYIKYRVEQVGDMIDRLFDTSA

Resource temporarily unavailable (4).
>gi|512998474|ref|XP_004860793.1| PREDICTED: exocyst complex component 7
isoform X2 [Heterocephalus glaber]
MIPPQEASARRREIEDKLKQEEETLSFIRDSLEKSDQLTKNMVSILSSFESRLMKLENSIIPVHKQTENL
QRLQENVEKTLSCLDHVISYYHVASDTEKIIREGPTGRLEEYLGSMAKIQKAVEYFQDNSPDSPELNKVK
LLFERGKESLESEFRSLMTRHSKVVSPVLILDLISGDEELEVQEDVALEHLPESVLQDVVRISLWLVEYG
RNQDFMNVYYQIRSSQLDRSIKGLKEHFRKSSSSSGVPYSPAIPNKRKDTPTKKPIKRPGTIRKAQNLLK
QYSQHGLDGKKGGSNLIPLEGRDDMLDVETDAYIHCVSAFVRLAQSEYQLLMDIIPEHHQKKTFDSLIQD
ALDGLMLEGENIVSAARKAIIRHDFSTVLTVFPILRHLKQTKPEFDQVLQGTAASTKNKLPGLITSMETI
GAKALEDFADNIKNDPDKEYNMPKDGTVHELTSNAILFLQQLLDFQETAGAMLASQVLGDTYNIPLDPRE
TSSSATSYSSEFSKRLLSTYICKVLGNLQLNLLSKSKVYEDPALSAIFLHNNYNYILKSLEKSELIQLVA
VTQKTAERSYREHIEQQIQTYQRSWLKVTDYISEKNLPVFQPGVKLRDKERQMIKERFKGFNDGLEELCK
IQKAWAIPDTEQRDKIRQAQKNIVKETYGAFLHRYSSVPFTKNPEKYIKYRVEQVGDMIDRLFDTSA

**Comment by antonkulaga** [comment]
@bbuchfink thank you very much. I fixed the line manually and everything works like a charm now!

**Comment by jieyalee0226** [comment]
hello, I have the same problem, but i don't konw how to find the invalid string in my data. please tell me. thank you.

**Comment by jinxin112233** [comment]
Hi, I encountered the same error reported. The error message is as follows:
`Error: Error reading input stream at line 25231: Invalid character (?) in sequence\n'`
This error message looks like it describes the location of the error, but it doesn't. My error message is on line 25231, but when I locate the line, I don't find any errors. Later I searched further in the text for the symbol(?) and found a large number of special characters .  I use `sed`  command to replaced the symbol like this: `sed 's/?//' file.pep > file_1.pep` The magic of this error is that the software reports the error location and the actual error location do not match.
